### PR TITLE
DOC: Update doc of parquet compression parameter

### PR DESCRIPTION
**Pull request by ChristofKaufmann**
This commit updates the interface documentation of [`pandas.DataFrame.to_parquet`](https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.to_parquet.html). The documentation of the parameter `compression` is not up-to-date compared to the implementation documentation of [`pandas.io.parquet.to_parquet`](https://github.com/pandas-dev/pandas/blob/fada873232ecb274f86c703770e41866084ae536/pandas/io/parquet.py#L399-L404). This commit copies the documentation of the implementation to the interface.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2813,8 +2813,12 @@ def to_parquet(
             ``io.parquet.engine`` is used. The default ``io.parquet.engine``
             behavior is to try 'pyarrow', falling back to 'fastparquet' if
             'pyarrow' is unavailable.
-        compression : {{'snappy', 'gzip', 'brotli', None}}, default 'snappy'
-            Name of the compression to use. Use ``None`` for no compression.
+        compression : {{'snappy', 'gzip', 'brotli', 'lz4', 'zstd', None}},
```
Comment (mroeschke) [inline pandas/core/frame.py:2816]:
```suggestion
        compression : {{'snappy', 'gzip', 'brotli', 'lz4', 'zstd', None}}, default 'snappy'
```

Then could you remove `default 'snappy'` from the next line?

Comment (ChristofKaufmann) [reply pandas/core/frame.py]: Sure.

**Comment by mroeschke** [comment]
Could you merge in the main branch? Appears to be a linting error as well.

**Comment by ChristofKaufmann** [comment]
Yes, now it has. It says "pandas/core/frame.py:2804:89: E501 Line too long (91 > 88 characters)". It refers to the [line](https://github.com/pandas-dev/pandas/pull/51554/files#diff-421998af5fe0fbc54b35773ce9b6289cae3e8ae607f81783af04ebf1fbcaf077R2804), where you requested to add the default value to get a consistent style. However, this is why the first line of this parameter description exceeds the maximum line width.

- We could move the default value back to the second line.
- We could replace the list of possible values by `str or None, default 'snappy'`. The text still describes possible values.
- We could ignore the failing check or somehow add an expected failure.
- You could extend the allowed line width.

What do you think is the best fix?

**Comment by mroeschke** [comment]
> We could replace the list of possible values by str or None, default 'snappy'. The text still describes possible values.

This option sounds best

**Comment by ChristofKaufmann** [comment]
Should be fine now.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @ChristofKaufmann

**Pull request merged**